src/main/java/com/treasure/hunt/utils/JavaFxUtils.java

```diff
@@ -1,12 +1,160 @@
 package com.treasure.hunt.utils;
 
+import com.treasure.hunt.analysis.SeriesStatisticObject;
+import javafx.collections.FXCollections;
+import javafx.embed.swing.SwingFXUtils;
+import javafx.geometry.Insets;
+import javafx.scene.control.*;
+import javafx.scene.image.WritableImage;
+import javafx.scene.layout.GridPane;
+import javafx.stage.FileChooser;
+import javafx.stage.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.javatuples.Septet;
 import org.jetbrains.annotations.NotNull;
 
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Slf4j
 public class JavaFxUtils {
 
     @NotNull
     public static String getVersionString() {
         String implementationVersion = JavaFxUtils.class.getPackage().getImplementationVersion();
         return implementationVersion == null ? "snapshot" : "v" + implementationVersion;
     }
+
+    public static File changeExtension(File f, String newExtension) {
+        int i = f.getName().lastIndexOf('.');
+        String name = f.getName().substring(0, i);
+        return new File(f.getParent() + "/" + name + newExtension);
+    }
+
+    public static void saveFile(Stage stage) {
+
+        WritableImage snapshot = stage.getScene().snapshot(null);
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setInitialFileName("plot.png");
+        File file = fileChooser.showSaveDialog(new Stage());
+
+        try {
+            ImageIO.write(SwingFXUtils.fromFXImage(snapshot, null), "png", file);
+
+        } catch (Exception e) {
+            log.error("Something went wrong when creating a file for plot!", e);
+            return;
+        }
+
+    }
```

Comment:
this really is not a descriptive name for a method, it seems to make a Screenshot not save an image